utility/src/test/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImplTest.kt

```diff
@@ -0,0 +1,281 @@
+import dagger.Provides
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.MockitoAnnotations
+import org.oppia.android.app.model.OppiaMetricLog
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.data.DataProvidersInjector
+import org.oppia.android.util.data.DataProvidersInjectorProvider
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.platformparameter.ENABLE_LANGUAGE_SELECTION_UI_DEFAULT_VALUE
+import org.oppia.android.util.platformparameter.EnableLanguageSelectionUi
+import org.oppia.android.util.platformparameter.LearnerStudyAnalytics
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG_DEFAULT_VALUE
+import org.oppia.android.util.platformparameter.SYNC_UP_WORKER_TIME_PERIOD_IN_HOURS_DEFAULT_VALUE
+import org.oppia.android.util.platformparameter.SplashScreenWelcomeMsg
+import org.oppia.android.util.platformparameter.SyncUpWorkerTimePeriodHours
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import org.robolectric.shadows.ShadowActivityManager
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TEST_PACKAGE_LABEL = "TEST_PACKAGE_LABEL"
+private const val TEST_APP_PATH = "TEST_APP_PATH"
+private const val TEST_PID = 1
+private const val TEST_FILE_NAME = "TEST_FILE_NAME"
+private const val TEST_FILE_CONTENT = "TEST_FILE_CONTENT"
+
+/** Tests for [PerformanceMetricsAssessorImpl]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PerformanceMetricsAssessorImplTest.TestApplication::class)
+class PerformanceMetricsAssessorImplTest {
+
+  @Inject
+  lateinit var performanceMetricsUtilsImpl: PerformanceMetricsAssessorImpl
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.initMocks(this)
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_setTotalMemory_returnsCorrectMemoryTier() {
+    val activityManager: ActivityManager =
+      ApplicationProvider.getApplicationContext<Application>()
+        .getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
```

Comment:
You could make this a bit simpler for this & other tests by leveraging fields, e.g.:

```kotlin
@Inject lateinit var context: Context

private val activityManager by lazy { context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager }
```

Reply:
done.